scipy/sparse/tests/test_base.py

```diff
@@ -2982,6 +2982,29 @@ def test_fancy_indexing_randomized(self):
         assert_raises(IndexError, S.__getitem__, (I_bad,J))
         assert_raises(IndexError, S.__getitem__, (I,J_bad))
 
+    def test_missized_masking(self):
+        M, N = 5, 10
+
+        B = asmatrix(arange(M * N).reshape(M, N))
+        A = self.spcreator(B)
+
+        # Content of mask shouldn't matter, only its size
+        row_long = np.ones(M + 1, dtype=bool)
+        row_short = np.ones(M - 1, dtype=bool)
+        col_long = np.ones(N + 2, dtype=bool)
+        col_short = np.ones(N - 2, dtype=bool)
+
+        assert_raises(IndexError, A.__getitem__, row_long)
+        assert_raises(IndexError, A.__getitem__, row_short)
```

Comment:
No need to modify existing tests, but for the new tests can we use the context manager version and match the message

Reply:
Done – looks a lot better 😄